mcp/sensors/star_camera_transmit.c

```diff
@@ -0,0 +1,396 @@
+        // fill out address info and return if it fails.
+        if ((returnval = getaddrinfo(socket_target->ipAddr, socket_target->port, &hints, &servinfo)) != 0) {
+            blast_err("getaddrinfo: %s\n", gai_strerror(returnval));
+            return NULL;
+        }
+        // now we make a socket with this info
+        for (servinfoCheck = servinfo; servinfoCheck != NULL; servinfoCheck = servinfoCheck->ai_next) {
+            if ((sockfd = socket(servinfoCheck->ai_family,
+             servinfoCheck->ai_socktype, servinfoCheck->ai_protocol)) == -1) {
+                perror("talker: socket");
+                continue;
+            }
+            break;
+        }
+        // check to see if we made a socket
+        if (servinfoCheck == NULL) {
+            // set status to 0 (dead) if this fails
+            blast_err("talker: failed to create socket\n");
+            return NULL;
+        }
+        // if we pass all of these checks then
+        // we set up the print statement vars
+        // need to cast the socket address to an INET still address
+        struct sockaddr_in *ipv = (struct sockaddr_in *)servinfo->ai_addr;
+        // then pass the pointer to translation and put it in a string
+        inet_ntop(AF_INET, &(ipv->sin_addr), ipAddr, INET_ADDRSTRLEN);
+        blast_info("IP target is: %s\n", ipAddr);
+        // now the "str" is packed with the IP address string
+        // first time setup of the socket is done
+        }
+        // Here we move on the main portion of the data comms
+        // check to see if a command packet has been packed
+        if (check_sc_send_commands(which_sc)) {
+            packet_status = prepare_packet(which_sc);
+        }
+        if (packet_status) {
+            if (!strcmp(scCommands->target, ipAddr)) {
+                packet_status = 0;
+                length = sizeof(*scCommands);
+                if ((bytes_sent = sendto(sockfd, scCommands, length, 0,
+                    servinfo->ai_addr, servinfo->ai_addrlen)) == -1) {
+                    perror("talker: sendto");
+                }
+                blast_info("%s sent packet to %s:%s\n", message_str, ipAddr, socket_target->port);
+                // clear the commands packet
+                blast_info("packet incharge is %d, real variable is %d\n", scCommands->inCharge, InCharge);
+                clear_structs(which_sc);
+            } else {
+                blast_err("Target destination differs from thread target.\n");
+            }
+        }
+        // Here we check to see if we're supposed to reset the socket
+        // If we want to reset we just free everything and tell it to do first time again
+        // otherwise we just sleep for 0.2 seconds and repeat
+        if (check_for_reset(which_sc)) {
+            blast_info("received reset command...");
+            first_time = 1;
+            freeaddrinfo(servinfo);
+            close(sockfd);
+        } else {
+            usleep(200000);
```

Comment:
No magic numbers, give this a name so a maintainer who wants to change it has some idea of why it's there/has the value it has. Comment above does this, but comments easily get out of date.